flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/JobSubmitHeaders.java

```diff
@@ -26,6 +27,8 @@
 import org.apache.flink.shaded.netty4.io.netty.handler.codec.http.HttpResponseStatus;
 
 /** These headers define the protocol for submitting a job to a flink cluster. */
+@Documentation.ExcludeFromDocumentation(
+        "This API is not exposed to the users, as ExecutionPlan is used only internally.")
```

Comment:
-> "This API is for Flink client only and should not be exposed to users, as it relies on internal classes."